src/controls/ObjectArrayControl.tsx

```diff
@@ -52,9 +51,12 @@ export function ObjectArrayControl({
   )
 
   const prevDataValue = usePreviousValue(data)
-  if (data === 0 && prevDataValue === null) {
-    addItemToList()
-  }
+
+  useEffect(() => {
+    if (data === 0 && prevDataValue === null) {
+      addItemToList()
+    }
+  })
```

Comment:
I would love to figure out why we have inconsistent state issues between `ObjectArrayControl` and `PrimitiveArrayControl`. Primitive doesn't need this `useEffect`.